applications/openshift/etcd/etcd_key_file/rule.yml

```diff
@@ -26,6 +26,20 @@ references:
 ocil_clause: 'the etcd client key file is not configured'
 
 ocil: |-
-    Run the following command on the master node(s):
-    <pre>$ grep ETCD_KEY_FILE=/etc/etcd/etcd.conf</pre>
-    Verify that there is a key file configured.
+    Run the following command:
+    <pre>oc get -nopenshift-etcd cm etcd-pod -oyaml | grep "\-\-key-file=/etc/kubernetes/static-pod-certs/secrets/etcd-all-serving/etcd-serving-NODE_NAME.key"</pre>
+    Verify that there is a certificate configured.
+
+warnings:
+    - general: |-
+        {{{ openshift_cluster_setting("/api/v1/namespaces/openshift-etcd/configmaps/etcd-pod") | indent(8) }}}
+
+template:
+    name: yamlfile_value
+    vars:
+        ocp_data: "true"
+        filepath: /api/v1/namespaces/openshift-etcd/configmaps/etcd-pod
+        yamlpath: ".data['pod.yaml']"
+        values:
+          - value: ".*--key-file=/etc/kubernetes/static-pod-certs/secrets/etcd-all-serving/etcd-serving-NODE_NAME.key \\.*"
```

Comment:
what happens with NODE_NAME here? I thought this would not be a substituted value. 

The --cert-file and --key-file checks can be a check that these flags are set with _any_ value, since they are required arguments.

Reply:
the literal value is NODE_NAME. I don't know why. But that's how the file actually looks like.